shm/Cargo.hpp

```diff
@@ -0,0 +1,33 @@
+#include <string>
+
+class Cargo {
+public:
+    Cargo() = default;
+    Cargo(std::string name, size_t amount, size_t basePrice)
+        : name_(name), amount_(amount), basePrice_(basePrice) {}
+
+    std::string getName() { return name_; }
+    size_t getAmount() { return amount_; }
+    size_t getBasePrice() { return basePrice_; }
```

Comment:
Shouldn't it be returned as a const and in case of string const & ?

Reply:
How about this ?
```
    std::string getName() const { return name_; }
    size_t getAmount() const { return amount_; }
    size_t getBasePrice() const { return basePrice_; }
```
I'm not sure if returning const& is good idea?

Reply:
Why not? strings can get rather large and returning by reference would save some time and memory(we are not copying an element). const ensures that user cannot modify it after returning it.

const after function signature just ensures that function itself isn't modifying anything.

Reply:
You're right. `const std::string&` makes sense. Thanks ;)